huaweicloud/services/cnad/data_source_huaweicloud_cnad_advanced_instances.go

```diff
@@ -0,0 +1,208 @@
+		Schema: map[string]*schema.Schema{
+			"package_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the package id.`,
+			},
+			"package_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the package name.`,
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: `Specifies the package instance type. Valid values are **cnad_pro**, **cnad_ip**,
+**cnad_ep**, **cnad_full_high**, **cnad_vic** and **cnad_intl_ep**.`,
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the region in which to query the data source.`,
+			},
+			"packages": {
+				Type:        schema.TypeList,
+				Elem:        instanceSchema(),
+				Computed:    true,
+				Description: `Indicates the list of the Advanced instances`,
+			},
+		},
+	}
+}
+
+func instanceSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"package_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the package id.`,
+			},
+			"package_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the package name.`,
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the region where the package belongs to.`,
+			},
+			"instance_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the instance type of the package.`,
+			},
+			"protection_type": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the protection type of the package.`,
+			},
+			"ip_num": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates ip num of the package.`,
+			},
+			"ip_num_now": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the current ip num of the package.`,
+			},
+			"protection_num": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the protection num of the package, value **9999** means unlimited times.`,
+			},
+			"protection_num_now": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the current protection num of the package.`,
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the created time.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceInstancesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+	var (
+		getAdvancedInstancesHttpUrl = "v1/cnad/packages"
+		getAdvancedInstancesProduct = "aad"
+	)
+	getAdvancedInstancesClient, err := cfg.NewServiceClient(getAdvancedInstancesProduct, region)
```

Comment:
the endpoint of AAD is global
```suggestion
	getAdvancedInstancesClient, err := cfg.NewServiceClient(getAdvancedInstancesProduct, "")
```

Reply:
OK